File: scratch/split_css.py
import re

with open('css/global.css', 'r', encoding='utf-8') as f:
    css = f.read()

# We will extract blocks that match specific top-level IDs.
# A block is roughly #ID_NAME ... { ... }
# We can find all top-level selectors by splitting by } that are at the root level.
# Actually, since cssutils is installed, let's use it! It's much safer.
import cssutils
import logging

cssutils.log.setLevel(logging.CRITICAL)

sheet = cssutils.parseString(css)

out_files = {
    'global.css': [],
    'home.css': [],
    'services.css': [],
    'about.css': [],
    'contact.css': [],
    'portfolio.css': [],
    'pricing.css': []
}

def get_target(selectorText):
    if '#homeSection' in selectorText or '#featureSection' in selectorText or '#workSection' in selectorText or '#premiumHeroSection' in selectorText or '#specialFeature' in selectorText or '#featureDetails' in selectorText:
        return 'home.css'
    elif '#serviceSection' in selectorText:
        return 'services.css'
    elif '#portfolioSection' in selectorText:
        return 'portfolio.css'
    elif '#pricingSection' in selectorText:
        return 'pricing.css'
    elif '#testimonialSection' in selectorText or '#aboutSection' in selectorText:
        return 'about.css'
    elif '#contactSection' in selectorText or '#contactArea' in selectorText or '#googleMap' in selectorText:
        return 'contact.css'
    else:
        return 'global.css'

for rule in sheet:
    if rule.type == rule.STYLE_RULE:
        target = get_target(rule.selectorText)
        out_files[target].append(rule.cssText)
    elif rule.type == rule.MEDIA_RULE:
        # Check inside media rule if there are specific targets
        # For simplicity, put media rules in global.css
        out_files['global.css'].append(rule.cssText)
    elif rule.type == rule.FONT_FACE_RULE or rule.type == rule.IMPORT_RULE:
        out_files['global.css'].append(rule.cssText)
    else:
        out_files['global.css'].append(rule.cssText)

import os
for fname, rules in out_files.items():
    if rules:
        with open('css/' + fname + '.new', 'w', encoding='utf-8') as f:
            f.write('\n\n'.join(rules))
        print(f"Wrote {fname}.new with {len(rules)} rules")

File: scratch/split_css.js
const fs = require('fs');

const cssPath = 'css/global.css';
const cssContent = fs.readFileSync(cssPath, 'utf8');

const lines = cssContent.split('\n');

const outFiles = {
    'global.css': [],
    'home.css': [],
    'services.css': [],
    'about.css': [],
    'contact.css': [],
    'portfolio.css': [],
    'pricing.css': []
};

// We will scan each line. If a line or block belongs to a section, put it there.
// A CSS block starts with a selector and ends with }

// Since the CSS is formatted with } on the same line or next line, we can just track if we are inside a rule.
// But actually, in the view_file, CSS rules look like:
// #homeSection .slider-box {
//     background: url(../img/home-bg.webp); ... }
//
// So each rule is clearly prefixed. We don't even need to track blocks perfectly if we just look at the prefix.
// If the line starts with or contains #homeSection, it's home.css. But what about properties inside the block?
// We need to track the CURRENT active block.

let currentTarget = 'global.css';
let insideBlock = false;
let braceCount = 0;

for (let i = 0; i < lines.length; i++) {
    let line = lines[i];
    
    // Count braces to know if we are inside a rule or media query
    let openBraces = (line.match(/\{/g) || []).length;
    let closeBraces = (line.match(/\}/g) || []).length;
    
    if (braceCount === 0 && line.trim().length > 0) {
        // We are at the root level, evaluating the selector
        if (line.includes('#homeSection') || line.includes('#featureSection') || line.includes('#workSection') || line.includes('#premiumHeroSection')) {
            currentTarget = 'home.css';
        } else if (line.includes('#serviceSection')) {
            currentTarget = 'services.css';
        } else if (line.includes('#portfolioSection')) {
            currentTarget = 'portfolio.css';
        } else if (line.includes('#pricingSection')) {
            currentTarget = 'pricing.css';
        } else if (line.includes('#testimonialSection') || line.includes('#aboutSection')) {
            currentTarget = 'about.css';
        } else if (line.includes('#contactSection')) {
            currentTarget = 'contact.css';
        } else if (line.includes('@media')) {
            // media queries go to global for now
            currentTarget = 'global.css';
        } else if (line.startsWith('/*')) {
            // keep currentTarget or go to global? Let's keep it.
        } else if (line.includes('#footerSection') || line.includes('#navigration')) {
            currentTarget = 'global.css';
        } else {
            currentTarget = 'global.css';
        }
    }
    
    outFiles[currentTarget].push(line);
    
    braceCount += openBraces;
    braceCount -= closeBraces;
    
    // Reset target when exiting block, but what about media queries?
    // Media queries have nested blocks. So braceCount > 0 inside media query.
    // This logic works! When braceCount hits 0, the next line is a new root selector.
}

for (const [file, content] of Object.entries(outFiles)) {
    if (content.length > 0) {
        fs.writeFileSync('css/' + file + '.new', content.join('\n'));
        console.log('Wrote', file + '.new', content.length, 'lines');
    }
}
